src/cata_tiles.cpp

```diff
@@ -887,6 +888,90 @@ void cata_tiles::draw( int destx, int desty, const tripoint &center, int width,
     SDL_RenderSetClipRect(renderer, NULL);
 }
 
+void cata_tiles::draw_minimap( int destx, int desty, const tripoint &center, int width, int height )
+{
+    if (!g) {
+        return;
+    }
+
+    int start_x = center.x - 55;
+    int start_y = center.y - 55;
+    int tiles_x = 110;
+    int tiles_y = 110;
+    int tile_size_x = std::max(width / tiles_x, 1);
+    int tile_size_y = std::max(height / tiles_y, 1);
+
+    // Center the drawn area within the total area.
+    int drawn_width = tiles_x * tile_size_x;
+    int drawn_height = tiles_y * tile_size_y;
+    int border_width = (width - drawn_width) / 2;
+    int border_height = (height - drawn_height) / 2;
+
+    auto &ch = g->m.access_cache( center.z );
+
+    // First draw terrain.
+    for( int y = 0; y < tiles_y; y++) {
+        for( int x = 0; x <= tiles_x; x++) {
+            tripoint p(start_x + x, start_y + y, center.z);
+
+            lit_level lighting = ch.visibility_cache[p.x][p.y];
+            SDL_Color color;
+            if(lighting == LL_DARK || lighting == LL_BLANK) {
+                color.r = 12;
+                color.g = 12;
+                color.b = 12;
+            } else {
+                int veh_part = 0;
+                vehicle *veh = g->m.veh_at( p, veh_part );
+                auto furniture = g->m.furn_at( p );
+                if (veh != nullptr) {
+                    color = cursesColorToSDL(veh->part_color(veh_part));
+                } else if ( g->m.has_furn( p ) ) {
+                    auto furniture = g->m.furn_at( p );
+                    color = cursesColorToSDL(furniture.color());
+                } else {
+                    auto terrain = g->m.ter_at( p );
+                    color = cursesColorToSDL(terrain.color());
+                }
+            }
+
+            SDL_SetRenderDrawColor(renderer, color.r, color.g, color.b, 255);
+            SDL_Rect rectangle;
+            rectangle.x = destx + border_width + x * tile_size_x;
+            rectangle.y = desty + border_height + y * tile_size_y;
+            rectangle.w = tile_size_x;
+            rectangle.h = tile_size_y;
+            SDL_RenderFillRect(renderer, &rectangle);
+        }
+    }
+
+    // Now draw critters over terrain.
+    for( int y = 0; y < tiles_y; y++) {
+        for( int x = 0; x <= tiles_x; x++) {
+            tripoint p(start_x + x, start_y + y, center.z);
+
+            lit_level lighting = ch.visibility_cache[p.x][p.y];
+            if(lighting != LL_DARK && lighting != LL_BLANK) {
+                const auto critter = g->critter_at( p, true );
+                if( critter != nullptr ) {
+                    SDL_Color color = cursesColorToSDL(critter->symbol_color());
+                    for(int xOffset = -tile_size_x; xOffset <= tile_size_x; xOffset++) {
```

Comment:
It works fine and all. But this loop should probably go into its own function (something like `draw_rhombus` or so) as it's currently a bit difficult to make out what this code does.